codegen/cmd.go

```diff
@@ -119,3 +166,70 @@ func (c Command) compileProtos(grp *render.Group) error {
 
 	return nil
 }
+
+func (c Command) generateBuild(operator model.Operator) error {
+	buildFiles, err := render.RenderBuild(operator)
+	if err != nil {
+		return err
+	}
+
+	writer := &writer.DefaultFileWriter{Root: c.BuildRoot}
+
+	if err := writer.WriteFiles(buildFiles); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c Command) buildPushImage(operator model.Operator) error {
+	image := operator.Deployment.Image
+	build := image.Build
+	if build == nil {
+		return nil
+	}
+
+	ldFlags := fmt.Sprintf("-X %v/pkg/version.Version=%v", c.moduleRoot, image.Tag)
+
+	// get the main package from the main directory
+	// assumes package == module name + main dir path
+	mainkPkg := filepath.Join(c.moduleName, filepath.Dir(build.MainFile))
+
+	binName := filepath.Join(c.BuildRoot, operator.Name)
+
+	err := util.GoBuild(util.GoCmdOptions{
+		BinName: binName,
+		Args: []string{
+			"-ldflags=" + ldFlags,
+			`-gcflags='all="-N -l"''`,
+		},
+		PackagePath: mainkPkg,
+		Env: []string{
+			"GO111MODULE=on",
+			"CGO_ENABLED=0",
+			"GOARCH=amd64",
+			"GOOS=linux",
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	defer os.Remove(binName)
+
+	fullImageName := fmt.Sprintf("%v/%v:%v", image.Registry, image.Repository, image.Tag)
+
+	buildCmd := docker.Command("build", "-t", fullImageName, c.BuildRoot)
+
+	if err := buildCmd.Run(); err != nil {
+		return err
+	}
+
+	if !build.Push {
+		return nil
+	}
+
+	pushCmd := docker.Command("push", fullImageName)
```

Comment:
Could we possibly create interfaces/mocks for these so that the code is more unit testable?